pkgs/development/ocaml-modules/topkg/default.nix

```diff
@@ -13,14 +17,15 @@ stdenv.mkDerivation rec {
     sha256 = "1slrzbmyp81xhgsfwwqs2d6gxzvqx0gcp34rq00h5iblhcq7myx6";
   };
 
-  nativeBuildInputs = [ opam ];
```

Comment:
Conceptually this line should be left as is because it changes the `opam` used for cross compilation.

Reply:
I’m not sure to understand your comment. I removed this line because no `opam` stuff from the environment (`$PATH`) is used during the build. Instead, the absolute path to `opam-installer` is resolved by `nix` when evaluating the `installPhase` attribute.

I’ve no clue how cross-compilation works with `nixpkgs` but I guess it’s not just a matter of having `opam` for the build architecture: how to get an OCaml cross-compiler? how to run it alongside `ocaml` for the build architecture (to run the build system, itself written in OCaml)?

Reply:
I know that cross compilation for OCaml is not implemented in Nixpkgs, so there is no practical reason to keep this line, but the way it works is that during native compilation `callPackage topkg/default.nix` will be called once, and the value of `nativeBuildInputs` will have no effect on the value of `opam`, but during cross compilation it will be called again, with the value of `nativeBuildInputs` from the first call affecting the value of `opam` in the second call.

Reply:
This should solved later, when the cross-compiling stuff has landed in nixpkgs.